server/sonar-server/src/test/java/org/sonar/server/notification/ws/AddActionTest.java

```diff
@@ -264,6 +268,18 @@ public void fail_when_using_branch_db_key() throws Exception {
     call(request.setProject(branch.getDbKey()));
   }
 
+  @Test
+  public void fail_when_user_does_not_have_BROWSE_permission() {
```

Comment:
this should be named "fail_when_user_does_not_have_USER_permission_on_private_project", as there should be no such failure on a public project (see other comment regarding this use case not being tested)